daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -265,6 +265,7 @@ private[validation] object Typing {
         }
       case (dfnName, dfn: DValue) =>
         Env(mod.languageVersion, world, ContextDefValue(pkgId, mod.name, dfnName)).checkDValue(dfn)
+      case (_, _: DTypeSyn) => // TODO #3616: check type synonyms
```

Comment:
And completing this missing piece!

Reply:
Happy to do this in a separate PR. We should still be able to test expansion of type synonyms without checking that the definition is well formed